dartagnan/src/main/java/com/dat3m/dartagnan/program/Program.java

```diff
@@ -19,9 +18,12 @@ public class Program {
 
     public enum SourceLanguage { LITMUS, LLVM, SPV }
 
+    public enum SpecificationType { EXISTS, FORALL, NOT_EXISTS, ASSERT }
+
     private String name;
-    private AbstractAssert spec;
-    private AbstractAssert filterSpec; // Acts like "assume" statements, filtering out executions
+    private SpecificationType specificationType = SpecificationType.ASSERT;
```

Comment:
It seems so. I kinda dislike the distinction between `forall` and `assert`, since the latter is conceptually an instance of the former. But I guess it still is an improvement of what we had before.